### PR TITLE
Exercise 8 of chapter 10 completed.

### PR DESCRIPTION
**Pull request by Htocki**

### CODE REVIEW

File: bin/10_8/src/Main.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+int main(int argc, char* argv[]) {
+    std::vector<std::string> file_names;
+
+    for (std::size_t i{ 1 }; i < argc; ++i) {
+        file_names.push_back(argv[i]);
+    }
+
+    std::ifstream 
+        file_one{ file_names[0] }, 
+        file_two{ file_names[1] };
```
Comment (sialiuk) [inline bin/10_8/src/Main.cpp]: The programme is crashed here in case argc <= 2.

File: bin/10_8/src/Main.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            char c;
+            file_one.get(c);
+            new_file.put(c);
+        }
+
+        while (file_two) {
+            char c;
+            file_two.get(c);
+            new_file.put(c);
+        }
```
Comment (sialiuk) [inline bin/10_8/src/Main.cpp]: Looks like code duplication.

File: bin/10_8/src/Main.cpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+        while (file_two) {
+            char c;
+            file_two.get(c);
+            new_file.put(c);
+        }   
+    }
+    else {
+        std::cerr << "An error occurred while opening files." << std::endl;
```
Comment (sialiuk) [inline bin/10_8/src/Main.cpp]: It's better to print the current error also.

File: bin/10_8/src/Main.cpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        file_two{ file_names[1] };
+
+    std::ofstream new_file ("NewFile.txt", std::ios_base::ate);
+
+    if (file_one.is_open() && 
+        file_two.is_open() &&
+        new_file.is_open()) 
+    {
+        while (file_one) {
+            char c;
```
Comment (sialiuk) [inline bin/10_8/src/Main.cpp]: Always initialize variables with default values.

**Review by sialiuk — CHANGES_REQUESTED** [review]

File: bin/10_8/src/Main.cpp (hunk 2 above)
Comment (Htocki) [reply bin/10_8/src/Main.cpp]: The way it is. The condition states that the contents of the first file are followed by the contents of the second, so I read the file after the file. I don’t see much point in complicating something, but if you have interesting thoughts on this section of the code, I’m all in the focus.

File: bin/10_8/src/Main.cpp (hunk 3 above)
Comment (Htocki) [reply bin/10_8/src/Main.cpp]: What do you have in mind?

**Review by sialiuk — APPROVED** [review]

**Pull request merged**